### PR TITLE
Handle global extensions that might have .vagrantplugins as well

### LINKED ISSUE #576: Update `.vagrantplugins` to handle global extensions
The [SequelPro](https://github.com/Chassis/SequelPro) extension needs to be updated to handle global extensions that use `.vagrantplugins`.

We'll need to modify this [section](https://github.com/Chassis/Chassis/blob/master/.vagrantplugins#L7).

We'll also need https://github.com/Chassis/SequelPro/pull/4 reviewed and merged before we fix this.

### PR DESCRIPTION
**Pull request by BronsonQuick**
Fixes #576

### CODE REVIEW

**Comment by BronsonQuick** [comment]
@ntwb Can you please pull down this branch and see if it works for you? Thank you Sir!

File: .vagrantplugins [hunk 1]
```diff
@@ -2,9 +2,16 @@ require_relative "puppet/chassis.rb"
...
+	# We also need to check the global plugins for .vagrantplugins as well.
+	global_ext_path = File.join(Dir.home, ".chassis", "extensions")
+	use_global_ext = Dir.exist?(global_ext_path) && ! Dir.empty?(global_ext_path)
+	if use_global_ext
+		@@global_ext_plugins = Dir.glob(global_ext_path + "/*/.vagrantplugins")
+	end
 
 	def self.load_extension_plugins()
 		subplugins = Dir.glob(File.join(@@dir, "extensions", "*", ".vagrantplugins"))
+		subplugins = @@global_ext_plugins
```
Comment (rmccue) [inline .vagrantplugins]: Is this an error if `@@global_ext_plugins` isn't set above?

File: .vagrantplugins [hunk 2]
```diff
@@ -2,9 +2,16 @@ require_relative "puppet/chassis.rb"
 
 module Chassis
 	@@dir = File.dirname(__FILE__)
+	# We also need to check the global plugins for .vagrantplugins as well.
+	global_ext_path = File.join(Dir.home, ".chassis", "extensions")
+	use_global_ext = Dir.exist?(global_ext_path) && ! Dir.empty?(global_ext_path)
+	if use_global_ext
+		@@global_ext_plugins = Dir.glob(global_ext_path + "/*/.vagrantplugins")
+	end
```
Comment (rmccue) [inline .vagrantplugins]: Maybe we should just move this block into the method below

File: .vagrantplugins [hunk 3]
```diff
@@ -5,6 +5,16 @@ module Chassis
 
 	def self.load_extension_plugins()
 		subplugins = Dir.glob(File.join(@@dir, "extensions", "*", ".vagrantplugins"))
+		# We also need to check the global plugins for .vagrantplugins as well.
+			global_ext_path = File.join(Dir.home, ".chassis", "extensions")
```
Comment (rmccue) [inline .vagrantplugins]: Indentation is off 😬

Comment (BronsonQuick) [reply .vagrantplugins]: fixed.

**Comment by BronsonQuick** [comment]
@rmccue Reckon you could have another look at this one please Sir?

File: .vagrantplugins [hunk 4]
```diff
@@ -5,6 +5,17 @@ module Chassis
...
 		subplugins = Dir.glob(File.join(@@dir, "extensions", "*", ".vagrantplugins"))
+		# We also need to check the global plugins for .vagrantplugins as well.
+		global_ext_path = File.join(Dir.home, ".chassis", "extensions")
+		use_global_ext = Dir.exist?(global_ext_path) && ! Dir.empty?(global_ext_path)
+
+		if use_global_ext
+			global_ext_plugins = Dir.glob(global_ext_path + "/*/.vagrantplugins")
+			# If we have global extensions that have .vagrantplugins then combine the arrays
+			if ! global_ext_plugins.empty?
+				subplugins += global_ext_plugins
```
Comment (rmccue) [inline .vagrantplugins]: Could we make this one line? `subplugins += global_ext_plugins unless global_ext_plugins.empty` I think?

Comment (BronsonQuick) [reply .vagrantplugins]: Fixed!

**Comment by BronsonQuick** [comment]
@rmccue I'm pinging you again because I'm a horrible person! 😁

**Pull request merged**